agent/uiserver/uiserver.go

```diff
@@ -251,15 +251,38 @@ func (h *Handler) renderIndex(cfg *config.RuntimeConfig, fs http.FileSystem) ([]
 	// have to match the encoded double quotes around the JSON string value that
 	// is there as a placeholder so the end result is an actual JSON bool not a
 	// string containing "false" etc.
-	re := regexp.MustCompile(`%22__RUNTIME_BOOL_[A-Za-z0-9-_]+__%22`)
+	re := regexp.MustCompile(`%22__RUNTIME_(BOOL|STRING)_([A-Za-z0-9-_]+)__%22`)
 
 	content = []byte(re.ReplaceAllStringFunc(string(content), func(str string) string {
-		// Trim the prefix and __ suffix
-		varName := strings.TrimSuffix(strings.TrimPrefix(str, "%22__RUNTIME_BOOL_"), "__%22")
-		if v, ok := tplData[varName].(bool); ok && v {
-			return "true"
+		// Trim the prefix and suffix
+		pair := strings.TrimSuffix(strings.TrimPrefix(str, "%22__RUNTIME_"), "__%22")
+		parts := strings.SplitN(pair, "_", 2)
+		switch parts[0] {
+		case "BOOL":
+			if v, ok := tplData[parts[1]].(bool); ok && v {
+				return "true"
+			}
+			return "false"
+		case "STRING":
+			if v, ok := tplData[parts[1]].(string); ok {
+				if bs, err := json.Marshal(v); err == nil {
+					return url.PathEscape(string(bs))
```

Comment:
Can you remind me why this escaping is done? At this point the json.Marshal should've escaped enough stuff right?

Reply:
Because this JSON is being printed directly into a URL encoded attribute on a meta tag in the index.html template...

Reply:
So for example the json encode will output `"foo"` but those quotes need to be URL encoded to prevent early termination of the meta tag attribute.

Reply:
Though probably for pedantic reasons this should be `html.EscapeString` it makes sense to continue urlpath-escaping something here if the ember side is also urlpath escaping the same thing later.

Reply:
RB made a great point offline that html encoding would be more appropriate which seems true however for whatever reason Ember (or some code in our front end at any rate) decided to use URL encoding and this is just keeping consistency!

Reply:
I actually wondered why we were using javascript escape/unescape and not encodeURIComponent/decodeURIComponent as they are ever so slightly different I think. I'm assuming that as long as we used the same encode/decode on both ends things would be fine. Is the golang you are using here the equivalent of JS `escape`?